fem/pgridfunc.cpp

```diff
@@ -225,23 +235,27 @@ void ParGridFunction::ExchangeFaceNbrData()
    MPI_Request *recv_requests = requests + num_face_nbrs;
    MPI_Status  *statuses = new MPI_Status[num_face_nbrs];
 
-   const double *h_data = this->HostRead();
-   for (int i = 0; i < send_data.Size(); i++)
+   auto d_data = this->Read();
```

Comment:
This breaks my application. **This** could be a alias, and the memory manager would throws an error about registering an alias.

Reply:
@v-dobrev , do you know what a good fix for this might be? I tried doing something like:
``` 
Vector mpi_data; 
mpi_data.MakeRef(*this,0);
```

but surprisingly it still gave me the same error.

Reply:
Actually, I'm wondering if this might require code changes on our end. I'll keep y'all in the loop.

Reply:
I close this for now since it seems to be on your side.